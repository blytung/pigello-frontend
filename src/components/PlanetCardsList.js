import { useEffect, useState } from "react";
import PlanetCard from "../components/PlanetCard";
const PlanetCardsList = ({ planets = [] }) => {
  const [isLoading, setIsLoading] = useState(true);
  useEffect(() => {
    if (planets.length) {
      setIsLoading(false);
    }
  }, [planets]);
  if (isLoading) {
    return (
      <div>
        <h1>Preparing for take off</h1>
      </div>
    );
  } else {
    return (
      <div className='plannet-wrapper'>
        {planets.map((planet, index) => (
          <PlanetCard key={index} planet={planet} />
        ))}
      </div>
    );
  }
};

export default PlanetCardsList;
