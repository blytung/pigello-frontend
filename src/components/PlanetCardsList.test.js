import React from "react";
import renderer from "react-test-renderer";

import PlanetCardsList from "./PlanetCardsList";

it("Renders when the array is empty", () => {
  const tree = renderer.create(<PlanetCardsList />).toJSON();
  expect(tree).toMatchSnapshot();
});

