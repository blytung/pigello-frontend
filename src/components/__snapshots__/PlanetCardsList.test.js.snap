// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders when the array is empty 1`] = `
<div>
  <h1>
    Preparing for take off
  </h1>
</div>
`;
